src/MapEditorContext/MapEditorContextReducer.ts

```diff
@@ -0,0 +1,53 @@
+import { randomInt } from "../utilities/randomInt";
+import {
+    AddPlatform,
+    DeletePlatform,
+    MapEditorContextActions,
+    SetActiveItem,
+    SetMousePosition
+} from "./MapEditorContextActions";
+import { MapEditorContextState } from "./MapEditorContextProvider";
+
+let globalId = 0;
+
+export const mapEditorContextReducer = (
+    state: MapEditorContextState,
+    action: MapEditorContextActions
+): MapEditorContextState => {
+    switch (action.type) {
+        case SetActiveItem:
+            return {
+                ...state,
+                activeItem: action.newItem
+            };
+        case AddPlatform:
+            return {
+                ...state,
+                platforms: [
+                    ...state.platforms,
+                    {
+                        x: randomInt(50, state.width - 100),
+                        y: randomInt(50, state.height - 100),
```

Comment:
Is this the platform's starting position? Why use random coordinates?

Reply:
It looks better if they're in a random position until we have a way to move them lol; the random values will be removed eventually